### ISSUE TITLE
Allow writing to "local source" which is externally synced

### ISSUE THREAD

**Issue by stevenbell** [bordaigorl/remy#18]
I've been using syncthing on my rM1 to keep a continuous backup of my files on my computer, and I can use reMy to view those files.  I'd like to be able to use reMy to upload files to that "backup" directory, which would then get synced over to the tablet.

1) Is there any reason not to do this?  Perhaps the JSON config file should specify that a local source can be written to, to avoid the case of accidentally overwriting a backup?

2) I'm happy to submit a PR; does this just require changes in the `LocalFileSource` class?  https://github.com/bordaigorl/remy/blob/9c5d6897c3ee40c1067aaff2d02e694854d90400/remy/remarkable/filesource.py#L101

**Comment by bordaigorl** [comment]
Hi @stevenbell sorry for the radio silence.
Have you solved this?
It would be pretty easy to do this:
- add a 'is_read_only' flag in the source options (default to true for local source)
- implement the `store` and `upload` functions (signature as found in `LiveFileSourceSSH`

**Comment by stevenbell** [comment]
Thanks, I'll take a shot at this and send a PR soon!

**Comment by bordaigorl** [comment]
Just a heads up: there's a new devel branch, you probably want to use that one as a base for the PR.

**Comment by martin-braun** [comment]
Any update on this? I like the idea a lot.

**Comment by martin-braun** [comment]
@stevenbell Thanks for the heads-up. It sounds promising so far. I think a good first step was to simply accept the requirement of a restart. A full restart takes too much time, a different launcher is a burden. 

Wouldn't it be enough to run `systemctl restart xochitl` on the device?

**Comment by stevenbell** [comment]
Yes, I think that would work.  The issue is that if stuff is syncing in the background there's not an obvious mechanism for triggering a `systemctl` command.  If I have to explicitly connect to the tablet, that sort of defeats the point of having the sync happen automatically in the background.

**Comment by martin-braun** [comment]
@stevenbell You are right. You would need something like a [hook](https://github.com/terminalnetwork/syncthing-hooks) that needs to run on the reMarkable. It would need to have a proper time out and it should not restart when a notebook is open. 

This is challenging, because the hook must only fire when syncthing is done and the device is idle, or it would at least delay the restart until that point.
